src/protocol/libp2p/kademlia/mod.rs

```diff
@@ -188,11 +187,23 @@ impl Kademlia {
 
         match self.peers.entry(peer) {
             Entry::Vacant(entry) => {
-                // TODO: verify that peer limit is respected
-                match self.pending_dials.remove(&peer) {
-                    Some(action) => match self.service.open_substream(peer) {
+                if let KBucketEntry::Occupied(entry) = self.routing_table.entry(Key::from(peer)) {
+                    entry.connection = ConnectionType::Connected;
+                }
```

Comment:
Are we not trying to insert the peer into the routing table on purpose? Doesn't make sense for outbound connections, but can't this be called when remote nodes discover us?

Reply:
I'm not sure I understand your comment. If the peer already exists in `peers`, we don't do anything because but if it doens't, we set its state to `Connected`. Am I missing something?

Reply:
I just thought we can learn the addresses of connecting peers (if the other peer initiated the connection) and insert them into the routing table. I'm not sure if this is a good idea, would bring any value on top of discovering peers via random walks, and wouldn't introduce a DoS vector.

Reply:
I think there is a certain "risk" in adding addresses reported via `TransportEvent::ConnectionEstablished` to routing table because we don't really know if those addresses are reachable and if they're not, then we advertise bad addresses to anyone who might get this peer in a `FIND_NODE` response.

Reply:
I agree, that was not the best idea.